### PR TITLE
GET /plans returns plans with same name & diff billingCycle

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner @ryanpbrewster 

we need to allow people to get not just their current custom plan, but custom plans with different billing cycles and the same `displayName` (read: category, name, we need a refactor)

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaidPlanRepo.scala [hunk 1]
```diff
@@ -54,4 +55,8 @@ class PaidPlanRepoImpl @Inject() (
     (for (row <- rows if row.state === PaidPlanStates.ACTIVE && row.kind.inSet(states)) yield row).list
   }
 
+  def getByDisplayName(displayName: String)(implicit session: RSession): Set[PaidPlan] = {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaidPlanRepo.scala:58]: How about `getByShortName`? or `getByGroupName`? Trying to push it in the eventual direction.

**Comment by andrewconner** [comment]
Nice, thanks for the tests!

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaidPlanRepo.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaidPlanRepo.scala:58]: I just didn't want to add confusion by introducing something inconsistent with the present. Are you against dropping `name` then renaming `displayName -> name`? That's what I'd like.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaidPlanRepo.scala:58]: Oh that's fine. Whatever you want to do, I think we agree on the end goal.

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/PaymentsControllerTest.scala [hunk 2]
```diff
@@ -46,9 +46,30 @@ class PaymentsControllerTest extends Specification with ShoeboxTestInjector {
...
+            editableFeatures = PaidPlanFactory.testPlanEditableFeatures, defaultSettings = PaidPlanFactory.testPlanSettings)
+        )
+        val standardMonthlyPlan = planRepo.save(
+          PaidPlan(kind = PaidPlan.Kind.NORMAL, name = Name[PaidPlan]("standard_monthly"), displayName = "Standard",
+            billingCycle = BillingCycle(1), pricePerCyclePerUser = DollarAmount(800),
+            editableFeatures = PaidPlanFactory.testPlanEditableFeatures, defaultSettings = PaidPlanFactory.testPlanSettings)
+        )
+        Seq(standardMonthlyPlan, standardBiannualPlan, standardAnnualPlan)
+      }
+    }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/PaymentsControllerTest.scala:74]: Can we use the `PaidPlanFactory`?

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/PaymentsControllerTest.scala [hunk 3]
```diff
@@ -86,40 +107,73 @@ class PaymentsControllerTest extends Specification with ShoeboxTestInjector {
...
+
+        val request2 = route.getAvailablePlans(publicId)
+        val response2 = controller.getAvailablePlans(publicId)(request2)
+
+        // don't get any custom plans
+        val plansByNameWithCustom = contentAsJson(response2)
+        (plansByNameWithCustom \ "plans" \ "Free").as[Seq[PaidPlanInfo]] === Seq(currentPlan.asInfo)
+        (plansByNameWithCustom \ "plans" \ "Standard").as[Seq[PaidPlanInfo]] === standardPlans.map(_.asInfo).sortBy(_.cycle.month)
+        (plansByNameWithCustom \ "current").as[PublicId[PaidPlan]] === PaidPlan.publicId(customPlans.head.id.get)
+        (plansByNameWithCustom \ "plans" \ "Enterprise").as[Option[Seq[PaidPlanInfo]]] === Some(customPlans.map(_.asInfo).sortBy(_.cycle.month))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/PaymentsControllerTest.scala]: Why is this testing `.as[Option[T]]`?

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/PaymentsControllerTest.scala]: Ah, I see, to get the same thing as above. I think `.asOpt` is more idiomatic.

**Comment by andrewconner** [comment]
Merge conflict, otherwise :horse: :rocket:

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/PaymentsControllerTest.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/PaymentsControllerTest.scala:74]: yup. just added a few factory methods by doing so, thanks.

**Comment by camhashemi** [comment]
the number of merge conflicts is too damn high

**Pull request merged**